File: 프로그래머스/2/12945. 피보나치 수/피보나치 수.java
class Solution {
    public int solution(int n) {
        
        int fn[] = new int[n+1];
        int div = 1234567;
        
        fn[0] = 0;
        fn[1] = 1;
        
        for(int i=2; i<=n; i++){
            fn[i] = (fn[i-2] + fn[i-1])%div;
        }
       
        return fn[n];
    }
}

File: 프로그래머스/2/12945. 피보나치 수/피보나치 수.js
function solution(n) {
    
    const fn = Array.from({length : n+1}).fill(0)
    const div = 1234567;
    
    fn[1] = 1;
    
    for(let i=2; i<=n; i++){
        fn[i] = (fn[i-2] + fn[i-1])%div;
    }

    return fn[n];
}